src/AccessibilityInsights.SharedUxTests/FileIssue/IssueReporterManagerTests.cs

```diff
@@ -143,11 +174,88 @@ public void SetIssueReporter_ReporterNonExistent_ReporterUnset()
             Assert.IsNull(IssueReporter.IssueReporting);
         }
 
-        private Mock<IIssueReporting> GetIssueReporterMock()
+        [TestMethod]
+        [Timeout(1000)]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void UpdateIssueReporterSettings_InputIsNull_ThrowsArgumentNullException()
+        {
+            ConfigurationModel configs = GetConfigurationModel(null);
+            IssueReporterManager repManager = new IssueReporterManager(configs, Enumerable.Empty<IIssueReporting>());
+
+            repManager.UpdateIssueReporterSettings(null);
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void UpdateIssueReporterSettings_ReporterDoesNotSupportGetSettings_DoesNothing()
+        {
+            ConfigurationModel configs = GetConfigurationModel(null);
+            IssueReporterManager repManager = new IssueReporterManager(configs, Enumerable.Empty<IIssueReporting>());
+            Mock<IIssueReporting> issueReportingMock = GetIssueReporterMock(
+                setStableIdentifier: false, expectRestoreConfig: false, supportGetSerializedSettings: false);
+
+            repManager.UpdateIssueReporterSettings(issueReportingMock.Object);
+
+            issueReportingMock.VerifyAll();
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void UpdateIssueReporterSettings_SettingNotInOriginalConfig_AddsToConfig()
         {
-            var issueReporterMock = new Mock<IIssueReporting>();
-            issueReporterMock.Setup(p => p.StableIdentifier).Returns(TestGuid);
-            issueReporterMock.Setup(p => p.RestoreConfigurationAsync(TestReporterConfigs)).Returns(new Task<IIssueResult>(() => { return null; }));
+            ConfigurationModel configs = GetConfigurationModel(null);
+            IssueReporterManager repManager = new IssueReporterManager(configs, Enumerable.Empty<IIssueReporting>());
+            Mock<IIssueReporting> issueReportingMock = GetIssueReporterMock(
+                expectRestoreConfig: false, supportGetSerializedSettings: true);
+
+            repManager.UpdateIssueReporterSettings(issueReportingMock.Object);
+
+            issueReportingMock.VerifyAll();
```

Comment:
should this test do something to verify the contents of `configs`?

Reply:
Added a check